tuist/tuist
# Fix missing linkable products for static frameworks with transitive precompiled dependencies

Resolves: https://github.com/tuist/tuist/issues/379

### Short description 📝

- When a static framework depends on a dynamic dependency (be it a native target or precompiled one) that dynamic dependency needs to be linked together with the static framework
- It seems we may have regressed or missed a case where:

```
App > StaticFramework > PrecompiledDynamicFramework
```

The should link `PrecompiledDynamicFramework`

- This change ensures we account for transitive dynamic precomplied frameworks of static products
- An existing fixture has been updated to include a similar scenario (for the Test target instead of an App, but the same logic applies)

### Test Plan 🛠:

- Run `tuist build && tuist generate --path fixtures/ios_app_with_transitive_framework`
- Open `fixtures/ios_app_with_transitive_framework/Workspace.xcworkspace`
- Build the scheme `Framework1Tests-Static`
- Verify the build succeeds

### Checklist ✅

- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- ~In case the PR introduces changes that affect users, the documentation has been updated.~
